### PR TITLE
[Workers] Update workers.yaml

### PR DESCRIPTION
**Pull request by Warfields**
Update changelog.

### CODE REVIEW

**Comment by Warfields** [comment]
Waiting on the preview service to update before moving this out of draft.

File: data/changelogs/workers.yaml [hunk 1]
```diff
@@ -2,6 +2,12 @@
 link: "/workers/platform/changelog/"
 productName: Workers
 entries:
+- publish_date: '2024-04-01'
+  description: |-
+    - The new unwrap_custom_thenables` compatibility flag enables workers to accept custom thenables in internal APIs that expect a promise (for instance, the `ctx.waitUntil(...)`` method).
+    - TransformStreams created with the TransformStream constructor now have a cancel algorithm that is called when the stream is canceled or aborted. This change is part of the implementation of the WHATWG Streams standard.
+    - The `nodejs-compat` mode now includes an implementation of the `MockTracker` API from `node:test`. This is not an implementation of the full `node:test` module, and mock timers are currently not included.
```
Comment (irvinebroque) [inline data/changelogs/workers.yaml]:
```suggestion
    - The [`nodejs_compat` compatibility flag](/workers/runtime-apis/nodejs/) now includes an implementation of the [`MockTracker` API from `node:test`](https://nodejs.org/api/test.html#class-mocktracker). This is not an implementation of the full `node:test` module, and mock timers are currently not included.
```

@jasnell for docs when back

File: data/changelogs/workers.yaml [hunk 2]
```diff
@@ -2,6 +2,12 @@
...
+    - Exceptions reported to tail workers now include a "stack" property containing the exception's stack trace, if available.
```
Comment (irvinebroque) [inline data/changelogs/workers.yaml]:
```suggestion
    - Exceptions reported to [Tail Workers](/workers/observability/logging/tail-workers/) now include a "stack" property containing the exception's stack trace, if available.
```

File: data/changelogs/workers.yaml [hunk 3]
```diff
@@ -2,6 +2,12 @@
 link: "/workers/platform/changelog/"
 productName: Workers
 entries:
+- publish_date: '2024-04-01'
+  description: |-
+    - The new unwrap_custom_thenables` compatibility flag enables workers to accept custom thenables in internal APIs that expect a promise (for instance, the `ctx.waitUntil(...)`` method).
```
Comment (irvinebroque) [inline data/changelogs/workers.yaml]:
```suggestion
    - The new `unwrap_custom_thenables` compatibility flag enables workers to accept custom thenables in internal APIs that expect a promise (for instance, the `ctx.waitUntil(...)`` method).
```

https://github.com/cloudflare/workerd/pull/1799 — this needs page in docs

File: data/changelogs/workers.yaml [hunk 4]
```diff
@@ -2,6 +2,12 @@
 link: "/workers/platform/changelog/"
 productName: Workers
 entries:
+- publish_date: '2024-04-01'
+  description: |-
+    - The new `unwrap_custom_thenables` compatibility flag enables workers to accept custom thenables in internal APIs that expect a promise (for instance, the `ctx.waitUntil(...)`` method).
```
Comment (irvinebroque) [inline data/changelogs/workers.yaml]: Can you add entry here? https://developers.cloudflare.com/workers/configuration/compatibility-dates/#change-history

Comment (Warfields) [reply data/changelogs/workers.yaml]: Sure thing!

Comment (Warfields) [reply data/changelogs/workers.yaml]: CC: @jasnell

Comment (Warfields) [reply data/changelogs/workers.yaml]: There was not much to go on for what the flag actually does.
https://github.com/cloudflare/workerd/pull/1799

**Review by irvinebroque — APPROVED** [review]

**Comment by Warfields** [comment]
Just learned I should have had the write permissions to merge from @jp4a50, which I don't. CC: @irvinebroque @deadlypants1973

**Pull request merged**